api/src/main/java/org/apache/iceberg/PartitionField.java

```diff
@@ -28,12 +28,14 @@ public class PartitionField implements Serializable {
   private final int fieldId;
   private final String name;
   private final Transform<?, ?> transform;
+  private final int schemaId;
 
-  PartitionField(int sourceId, int fieldId, String name, Transform<?, ?> transform) {
```

Comment:
I think we want to keep this constructor, and add another one to avoid breaking the API.